src/components/UI/Loader/Loader.tsx

```diff
@@ -0,0 +1,39 @@
+import React from "react";
+import { Loader as MantineLoader, useMantineTheme } from "@mantine/core";
+import { loaderVariants } from "./styles";
+
+type LoaderSize = "small" | "medium" | "large" | "extraLarge";
+
+interface LoaderProps {
+  size?: LoaderSize;
+  variant?: keyof typeof loaderVariants;
+  className?: string;
+}
+const sizeMapper: Record<LoaderSize, "xs" | "sm" | "md" | "lg" | "xl"> = {
+  small: "sm",
+  medium: "md",
+  large: "lg",
+  extraLarge: "xl",
+};
+
+export const Loader: React.FC<LoaderProps> = ({
+  size = "medium",
+  variant = "primary",
+  className, 
+  ...props
+}) => {
+  const variantStyle = loaderVariants[variant] || loaderVariants.primary;
+
+  return (
+    <div style={{
```

Comment:
Please no inline styles, you can create `styles.ts` file just like we are having inside the entire project